tests/libs/util/wer_report.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#define WIN32_LEAN_AND_MEAN // Exclude rarely-used stuff from Windows headers
+#include <Windows.h>
+#include <errorrep.h>
+#include <stdexcept>
+#include <string>
+#include <werapi.h>
+
+#pragma comment(lib, "wer.lib")
+/**
+ * @brief Class to automatically capture WER Report / Triage crash dump on fatal application exception.
+ * Exceptions are only logged if the environment variable EBPF_ENABLE_WER_REPORT is set to "yes".
+ *
+ */
+class _wer_report
+{
+  public:
+    _wer_report() : vectored_exception_handler_handle(nullptr)
+    {
+        unsigned long guaranteed_stack_size = static_cast<unsigned long>(minimum_stack_size_for_wer);
+        char* buffer = nullptr;
+        size_t size = 0;
+        _dupenv_s(&buffer, &size, environment_variable_name);
+        if (size == 0 || !buffer || _stricmp(environment_variable_value, buffer) != 0) {
+            free(buffer);
+            return;
+        }
+        free(buffer);
+
+        if (!SetThreadStackGuarantee(&guaranteed_stack_size)) {
+            throw std::runtime_error("SetThreadStackGuarantee failed");
+        }
+        vectored_exception_handler_handle = AddVectoredExceptionHandler(TRUE, _wer_report::vectored_exception_handler);
+    }
+    ~_wer_report()
+    {
+        if (vectored_exception_handler_handle)
+            RemoveVectoredExceptionHandler(vectored_exception_handler_handle);
+    }
+
+  private:
+    static constexpr const char environment_variable_name[] = "EBPF_ENABLE_WER_REPORT";
```

Comment:
https://github.com/microsoft/ebpf-for-windows/blob/main/docs/DevelopmentGuide.md#naming-conventions says:
> Prefix names with `_` to indicate internal and private fields or methods (e.g., `_internal_field`, `_internal_method()`).